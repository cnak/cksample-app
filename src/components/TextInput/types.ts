import ReactNative from 'react-native';

export interface TextInputProps extends ReactNative.TextInputProps {
    width?: string;
}

export interface TextInputInternalState {
    showFocus: boolean;
}
