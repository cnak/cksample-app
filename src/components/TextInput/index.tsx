import React, { Component } from 'react';
import styled from 'styled-components/native';

import { rem } from '../Typography';
import { YELLOW } from '../colors';

import { TextInputInternalState, TextInputProps } from './types';

const StyledInput = styled.TextInput`
    border: 2px solid black;
    font-size: ${rem(1)}px;
    height: 40px;
    padding: 5px;
    width: 100%;
    line-height: ${rem(1.25 * 1)}px;
`;

export const OUTLINE_WIDTH = 3;

const OutlineView = styled.View<{ showFocus: boolean; width: string }>`
    border-width: ${OUTLINE_WIDTH}px;
    border-color: ${props => (props.showFocus ? YELLOW : 'transparent')};
    width: ${props => props.width};
`;

export default class TextInput extends Component<TextInputProps, TextInputInternalState> {
    constructor(props: any) {
        super(props);
        this.state = { showFocus: false };
    }

    public render() {
        const { showFocus } = this.state;
        const { width = '100%' } = this.props;
        return (
            <OutlineView showFocus={showFocus} width={width}>
                <StyledInput
                    testID="TextInput"
                    onFocus={this.onFocus(true)}
                    onBlur={this.onFocus(false)}
                    spellCheck={false}
                    autoCorrect={false}
                    {...this.props}
                />
            </OutlineView>
        );
    }

    private onFocus = (bool: boolean) => () => {
        this.setState({ showFocus: bool });
        return true;
    };
}
