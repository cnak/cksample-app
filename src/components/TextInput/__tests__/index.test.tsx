import React from 'react';
import renderer from 'react-test-renderer';

import TextInput from '../.';

describe('<TextInput>', () => {
    it('matches snapshot', () => {
        const tree = renderer.create(<TextInput>This is a test</TextInput>).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
