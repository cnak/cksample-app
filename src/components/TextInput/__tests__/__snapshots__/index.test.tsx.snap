// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextInput> matches snapshot 1`] = `
<View
  showFocus={false}
  style={
    Array [
      Object {
        "borderColor": "transparent",
        "borderWidth": 3,
        "width": "100%",
      },
    ]
  }
  width="100%"
>
  <TextInput
    allowFontScaling={true}
    autoCorrect={false}
    onBlur={[Function]}
    onFocus={[Function]}
    rejectResponderTermination={true}
    spellCheck={false}
    style={
      Array [
        Object {
          "borderColor": "black",
          "borderStyle": "solid",
          "borderWidth": 2,
          "fontSize": 16,
          "height": 40,
          "lineHeight": 20,
          "paddingBottom": 5,
          "paddingLeft": 5,
          "paddingRight": 5,
          "paddingTop": 5,
          "width": "100%",
        },
      ]
    }
    testID="TextInput"
    underlineColorAndroid="transparent"
  >
    This is a test
  </TextInput>
</View>
`;
